File: devorbit-api/src/main/java/vn/edu/uit/devorbit_api/dto/student/GroupPlanMemberResponse.java
package vn.edu.uit.devorbit_api.dto.student;

import vn.edu.uit.devorbit_api.constant.MemberStatus;
import vn.edu.uit.devorbit_api.entity.GroupPlanMember;

public record GroupPlanMemberResponse(
    Long id,
    Long groupPlanId,
    String studentCode,
    String status,
    String invitedAt,
    String respondedAt
) {
    public static GroupPlanMemberResponse from(GroupPlanMember m) {
        return new GroupPlanMemberResponse(
            m.getId(),
            m.getGroupPlan().getId(),
            m.getStudentCode(),
            m.getStatus().name(),
            m.getInvitedAt() != null ? m.getInvitedAt().toString() : null,
            m.getRespondedAt() != null ? m.getRespondedAt().toString() : null
        );
    }
}

File: devorbit-mobile/app/src/main/java/vn/edu/uit/devorbit/mobile/data/remote/dto/GroupPlanMemberResponse.kt
package vn.edu.uit.devorbit.mobile.data.remote.dto

data class GroupPlanMemberResponse(
    val id: Long,
    val groupPlanId: Long,
    val studentCode: String,
    val status: String,
    val invitedAt: String?,
    val respondedAt: String?
)
